test/src/unit-QueryCondition.cc

```diff
@@ -854,3 +854,160 @@ TEST_CASE(
 
   free(values);
 }
+
+TEST_CASE(
+    "QueryCondition: Test empty string", "[QueryCondition][empty_string]") {
+  const std::string field_name = "foo";
+  const uint64_t cells = 10;
+  const char* fill_value = "ac";
+  const Datatype type = Datatype::STRING_ASCII;
+  bool var_size = GENERATE(true, false);
+  bool nullable = GENERATE(true, false);
+  QueryConditionOp op = GENERATE(QueryConditionOp::NE, QueryConditionOp::EQ);
+
+  // Initialize the array schema.
+  ArraySchema array_schema;
+  Attribute attr(field_name, type);
+  REQUIRE(attr.set_nullable(nullable).ok());
+  REQUIRE(attr.set_cell_val_num(var_size ? constants::var_num : 2).ok());
+
+  if (!nullable) {
+    REQUIRE(attr.set_fill_value(fill_value, 2 * sizeof(char)).ok());
+  }
+
+  REQUIRE(array_schema.add_attribute(&attr).ok());
+  Domain domain;
+  Dimension dim("dim1", Datatype::UINT32);
+  uint32_t bounds[2] = {1, cells};
+  Range range(bounds, 2 * sizeof(uint32_t));
+  REQUIRE(dim.set_domain(range).ok());
+  REQUIRE(domain.add_dimension(&dim).ok());
+  REQUIRE(array_schema.set_domain(&domain).ok());
+
+  // Initialize the result tile.
+  ResultTile result_tile(0, 0, &domain);
+  result_tile.init_attr_tile(field_name);
+
+  ResultTile::TileTuple* const tile_tuple = result_tile.tile_tuple(field_name);
+
+  var_size = array_schema.attribute(field_name)->var_size();
+  nullable = array_schema.attribute(field_name)->nullable();
+  Tile* const tile =
+      var_size ? &std::get<1>(*tile_tuple) : &std::get<0>(*tile_tuple);
+
+  REQUIRE(tile->init_unfiltered(
+                  constants::format_version,
+                  type,
+                  2 * cells * sizeof(char),
+                  2 * sizeof(char),
+                  0)
+              .ok());
+
+  char* values = static_cast<char*>(malloc(sizeof(char) * 2 * cells));
+  for (uint64_t i = 0; i < cells; ++i) {
+    values[i * 2] = 'a';
+    values[(i * 2) + 1] = 'a' + static_cast<char>(i);
+  }
```

Comment:
should `values` include the empty string?

Reply:
(cc @bdeng-xt -- let's confirm that the test data is testing what we need to test here)

Reply:
I am creating a test data with empty strings. var_size will be always be true for the test data